### PR TITLE
Use x86 crossbuild for cross-architecture component on x64-linux

### PR DESCRIPTION
**Pull request by hseok-oh**
Fix build.sh and build system generator to use x86 cross-build system on x64-linux
- x86-host/ARM32-target cross-architecture component build on x64 linux
- use x86 root file system and cross buildtool

related issue: #8685

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -793,17 +793,29 @@ build_native $__SkipCoreCLR "$__BuildArch" "$__IntermediatesDir" "$__ExtraCmakeA
 # Build cross-architecture components
 if [ $__CrossBuild == 1 ]; then
     __SkipCrossArchBuild=1
+    TARGET_ROOTFS=""
+    export CROSSCOMPILE=0
     if [ $__DoCrossArchBuild == 1 ]; then
         # build cross-architecture components for x86-host/arm-target
         if [[ "$__BuildArch" == "arm" && "$__CrossArch" == "x86" ]]; then
             __SkipCrossArchBuild=0
+            if [[ "$__HostArch" == x64 ]]; then
```
Comment (janvorli) [inline build.sh]: Why is this specific to x64 host? What if the `__HostArch` was e.g. arm64? Should it rather check if the `__HostArch` is not x86?

Comment (hseok-oh) [inline build.sh]: @janvorli Is it need to consider cross-architecture component in host-arm64/target-arm?

**Comment by hseok-oh** [comment]
@janvorli PTAL

File: build.sh (hunk 1 above)
Comment (janvorli) [inline build.sh]: @hseok-oh never mind, I have just realized that when `_CrossArch` is x86, the `_HostArch` can be only x86 or x64.

File: build.sh [hunk 2]
```diff
@@ -266,6 +266,46 @@ build_native()
...
+    else
+        # not supported
+        return
+    fi    
+    
+    export __CMakeBinDir="$__CrossComponentBinDir"
+    export CROSSCOMPONENT=1
+    if [ $CROSSCOMPILE == 1 ]; then
+        TARGET_ROOTFS="$ROOTFS_DIR"
+        if [ -n "$CAC_ROOTFS_DIR" ]; then
```
Comment (hseok-oh) [inline build.sh]: @janvorli I've update it

File: src/pal/tools/gen-buildsys-clang.sh [hunk 3]
```diff
@@ -128,18 +128,17 @@ if [[ -n "$LLDB_INCLUDE_DIR" ]]; then
...
+        echo "ROOTFS_DIR not set for crosscompile"
+        exit 1
+    fi
+    if [[ -z $CONFIG_DIR ]]; then
+        CONFIG_DIR="$1/cross/$build_arch"
     fi
+    cmake_extra_defines="$cmake_extra_defines -C $CONFIG_DIR/tryrun.cmake"
+    cmake_extra_defines="$cmake_extra_defines -DCMAKE_TOOLCHAIN_FILE=$CONFIG_DIR/toolchain.cmake"
 fi
 if [ "$build_arch" == "arm-softfp" ]; then
```
Comment (janvorli) [inline src/pal/tools/gen-buildsys-clang.sh]: This should be armel now that I've merged the change that changed the usage of arm-softfp to armel, right?

Comment (hseok-oh) [inline src/pal/tools/gen-buildsys-clang.sh]: Yes. But the PR change arm-softfp to armel already modify this. 
I'll merge newer master branch and clear conflict.

**Pull request merged**